shell/platform/linux/fl_external_texture_gl.h

```diff
@@ -0,0 +1,90 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_EXTERNAL_TEXURE_GL_H
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_EXTERNAL_TEXURE_GL_H
+
+#include <glib-object.h>
+
+#include "flutter/shell/platform/embedder/embedder.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_texture_registrar.h"
+
+G_BEGIN_DECLS
+
+G_DECLARE_FINAL_TYPE(FlExternalTextureGl,
+                     fl_external_texture_gl,
+                     FL,
+                     EXTERNAL_TEXTURE_GL,
+                     GObject)
+
+/**
+ * FlExternalTextureGl:
+ *
+ * #FlExternalTextureGl is an abstraction over OpenGL textures.
+ */
+
+/**
+ * fl_external_texture_gl_new:
+ * @texture_callback: An #FlTextureCallback.
+ * @user_data: A void*.
+ *
+ * Creates a new #FlExternalTextureGl.
+ *
+ * Returns: a new #FlExternalTextureGl.
+ */
+FlExternalTextureGl* fl_external_texture_gl_new(
+    FlTextureCallback texture_callback,
+    void* user_data);
```

Comment:
This should definitely be part of the interface but not sure how to make it generic based on `FlTextureCallback` returning different values. One way would be to change the output buffer of `FlTextureCallback` to a `void*`, then the subclass defines what is supposed to be output.